src/middlewares/condom.rs

```diff
@@ -0,0 +1,1305 @@
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("CondomMiddleware")
+            .field("limits", &self.limits)
+            .finish()
+    }
+}
+
+impl ModuleMiddleware for CondomMiddleware {
+    /// This method is called before applying middleware on functions,
+    /// and it will enforce the export limits on the module.
+    fn transform_module_info(&self, module_info: &mut ModuleInfo) -> Result<(), MiddlewareError> {
+        // Check the export limits and return an error if exceeded
+        if let Some(max_exports) = self.limits.max_exports {
+            if module_info.exports.len() > max_exports {
+                return Err(MiddlewareError::new(
+                    "CondomMiddleware",
+                    "The WASM file has too many exports. Blocked to prevent compile-time memory bloat",
+                ));
+            }
+        }
+
+        // Check the number of function limits
+        if let Some(max_functions) = self.limits.max_functions {
+            if module_info.functions.len() > max_functions {
+                return Err(MiddlewareError::new(
+                    "CondomMiddleware",
+                    "The WASM file has too many functions. Blocked to prevent compile-time memory bloat",
+                ));
+            }
+        }
+
+        // Check the signature limits i.e. the number of arguments a function
+        // can take.
+        // NOTE, at the moment (2024-09-30) :
+        // - functions can returns only one value, see:
+        // https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format#signatures_and_parameters
+        // - tweaking test_condom_middleware_exceeds_args_limit_max(), one can
+        //   see that wasmer limits number of arguments to 1024
+        if let Some(max_signature_len) = self.limits.max_signature_len {
+            for signature in module_info.signatures.values() {
+                if signature.params().len() + signature.results().len() > max_signature_len {
+                    return Err(MiddlewareError::new(
+                        "CondomMiddleware",
+                        "The WASM file contains a function with too many parameters and return values. Blocked to prevent compile-time memory bloat",
+                    ));
+                }
+            }
+        }
+
+        // Check the name limit
+        if let Some(max_name_len) = self.limits.max_name_len {
+            if let Some(name) = &module_info.name {
+                if name.len() > max_name_len {
+                    dbg!(max_name_len, name.len());
+                    return Err(MiddlewareError::new(
+                        "CondomMiddleware",
+                        "The WASM file has a too long name. Blocked to prevent compile-time memory bloat",
+                    ));
+                }
+            }
+        }
+
+        // Check the import limits
+        #[cfg_attr(feature = "enable-serde", serde(with = "serde_imports"))]
```

Comment:
Why is this needed?

Reply:
Can't remember and it seems useless, I'm removing it.